src/Illuminate/Auth/Authenticatable.php

```diff
@@ -38,7 +38,17 @@ public function getAuthIdentifier()
      */
     public function getAuthPassword()
     {
-        return $this->password;
+        return $this->{$this->getAuthPasswordField()};
+    }
+
+    /**
+     * Get the password field for the user.
+     *
+     * @return string
+     */
+    public function getAuthPasswordField()
```

Comment:
What's the benefit of adding additional method?

If field name is not "password" you should change `return password;` to `return custom;` and without this method you need to just change `return $this->password;` to `return $this->custom;` so this change doesn't give much benefit. 

And in fact you needed some more logic you need to change `getAuthPassword` after all so in my opinion change in this file doesn't improve anything.

Reply:
Sorry, @mnabialek I forgot to add one file which uses this method. Please see the EloquentUserProvider I just committed.